distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/KadActionsBuilder.java

```diff
@@ -0,0 +1,309 @@
+package com.gruppo1.distributednetworkmanager;
+
+import com.dezen.riccardo.networkmanager.StringResource;
+import com.dezen.riccardo.smshandler.SMSPeer;
+
+import java.util.Random;
+
+/**
+ * @author Niccolo' Turcato
+ * Builds KadAction objs with randomized ids (1-999)
+ * <p>
+ * TODO: add detailed description (summary from De Zen's Paper)
+ */
+public class KadActionsBuilder implements ActionsBuilder<KadAction, SMSPeer, StringResource, Node<BinarySet>> {
+
+
+    private static int MAX_ACTION_ID;
+    private static final String PAYLOAD_IGNORED = "0w0";
+    private static final String ILLEGAL_MAX_ID_MSG = "Can't initialize Builder, the defined max ID isn't compatible with class KadActions";
+    public static final String RESOURCE_SEPARATOR = KadAction.RESOURCE_SEPARATOR;
+
+    public static final KadAction INVALID_ACTION = KadAction.INVALID_KAD_ACTION;
+
+    /**
+     * Constructor for this builder, defines a new max limit for IDs, min limit remains the one defined in KadAction
+     *
+     * @param maxActionID max value for action IDs that this builder will accept, must be a valid KadAction ID
+     */
+    public KadActionsBuilder(int maxActionID) {
+        if (maxActionID >= KadAction.MIN_ID && maxActionID <= KadAction.MAX_ID)
+            MAX_ACTION_ID = maxActionID;
+        else
+            throw new IllegalArgumentException(ILLEGAL_MAX_ID_MSG);
+    }
+
+    /**
+     * Default constructor, builder will accept all valid IDs, (as indicated in KadAction)
+     */
+    public KadActionsBuilder() {
+        MAX_ACTION_ID = KadAction.MAX_ID;
+    }
+
+    private boolean isValidID(int id) {
+        return id >= KadAction.MIN_ID && id <= MAX_ACTION_ID;
+    }
+
+    /**
+     * @param peer     the peer that will be pinged
+     * @param actionID the unique ID of the action
+     * @return if the given ID is valid: an action built to execute the ping command to the peer, otherwise returns an INVALID_ACTION
+     */
+    public KadAction buildPing(int actionID, SMSPeer peer) {
+        if (isValidID(actionID))
+            return new KadAction(
+                    peer,
+                    KadAction.ActionType.PING,
+                    actionID,
+                    1, 1,
+                    KadAction.PayloadType.IGNORED,
+                    PAYLOAD_IGNORED
+            );
+        else return INVALID_ACTION;
+    }
+
+    /**
+     * @param request the ping request received from a peer (that will be the receiver for the response)
+     * @return a response action built to respond to the received ping command from the peer, or an invalid KAD action if the given action wasn't coherent with this building method (actionType)
+     */
+    public KadAction buildPingAnsw(KadAction request) {
```

Comment:
Since it's just 2 letters I'd just go for "Answer"

Reply:
You're right, but I liked this more, anyway I should change the interface too, dunno